### PR TITLE
(chore) add `.js` extension to import statements

### PR DESCRIPTION
**Pull request by aduh95**
Add file extension to all `import` specifiers in `./src/` files.
This is useful to run the files straight from source with a web browser, Node.js ESM or Deno.

Refs: https://github.com/highlightjs/highlight.js/issues/2600#issuecomment-642282239

### CODE REVIEW

**Comment by yyyc514** [comment]
What about the requires inside src/languages?   Are all those already good?

**Comment by yyyc514** [comment]
Is this some way to make rollup REQUIRE this?  Any idea?  This isn't going to magically keep itself maintained...

File: .eslintrc.js [hunk 1]
```diff
@@ -36,6 +36,11 @@ module.exports = {
       "indent": ["error", 2, {"VariableDeclarator":2}],
       // seems like a good idea not to use explicit undefined
       "no-undefined": "error",
+      // ensure import specifier contains file extension
+      "import/extensions": ["error", "always"],
+      // make sure there is no Node.js specific API slipping into the source files
+      "import/no-nodejs-modules": "error",
+      "import/no-commonjs": "error",
```
Comment (yyyc514) [inline .eslintrc.js:43]: Do we need to add a npm library for these?  Where can I find the docs?  I don't see these on the main ESLIst list of rules?

Comment (aduh95) [reply .eslintrc.js:43]: The library is already there, we just don't use it yet for some reason:

https://github.com/highlightjs/highlight.js/blob/1936dbd59dcd4de7391a9ea0413a68906fb04c14/package.json#L50

Comment (yyyc514) [reply .eslintrc.js:43]: I think these (or at least the nodejsone) should be moved to `overrides` and scope to `src/*`, no?  We definitely DO use nodejss modules in tests.  Or the rule could just be removed; I don't see a big risk of Node.js modules accidentally slipping into our client libraries.

**Pull request merged**